### ISSUE TITLE
[CKV_AWS_338] 1 year log retention period should be justified better

### ISSUE THREAD

**Issue by srgoni** [bridgecrewio/checkov#5470]
**Describe the issue**
CKV_AWS_338 may report false positives when a shorter log retention is chosen than 1 year.

**Examples**
```hcl
resource "aws_cloudwatch_log_group" "v2" {
  name = "test-log-group"
  retention_in_days = 180
}
```

**Version (please complete the following information):**
2.4.5

**Additional context**
The check source (https://github.com/bridgecrewio/checkov/blob/main/checkov/terraform/checks/resource/aws/CloudWatchLogGroupRetentionYear.py) mentions NIST SP 800-53 Rev. 5, but this document states:

> AU-11 AUDIT RECORD RETENTION
Control: Retain audit records for _[Assignment: organization-defined time period consistent with
records retention policy]_ to provide support for after-the-fact investigations of incidents and to
meet regulatory and organizational information retention requirements.

There is no mention of a fixed 1-year retention period.

On the other hand, there is an AWS Security Hub control that specifies a 1-year period: https://docs.aws.amazon.com/securityhub/latest/userguide/cloudwatch-controls.html#cloudwatch-16

The check should reference this security control in addition to the NIST specification, or possibly allow customizing the retention period according to organization policies (if this is possible).

**Comment by gruebel** [comment]
hey @srgoni thanks for reaching out.

You are right this check is tightly related to the Security Hub control. We usually try to be aligned with the recommendation of the cloud provider itself. We typically add proper information about the reasoning of a check into its guidelines, but this can take some time.

None of our checks are adjustable, but you can copy them, adjust it and reference it via the external dir flag. We also have check `CKV_AWS_66` which just looks for the existence of the retention attribute with any value.
